Source/Test/Rewriting/RewritingEngine.cs

```diff
@@ -74,9 +79,10 @@ public class RewritingEngine
         private readonly Dictionary<string, AssemblyNameDefinition> RewrittenAssemblies;
 
         /// <summary>
-        /// List of assemblies to be rewritten.
+        /// List of assembly names to be rewritten, these are strong names like:
+        /// "Microsoft.AspNetCore.Mvc.Core, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60".
         /// </summary>
-        private Queue<string> Pending = new Queue<string>();
+        private HashSet<string> AssemblyNameScope = new HashSet<string>();
```

Comment:
So we now assume that all possible assemblies that can be rewritten for say a given program are rewritten in one go, i.e. one invocation of `coyote rewrite foo.json`, whereas before we didn't have this assumption. I think this is fair to assume, after all when you compile a .NET project you are supposed to have already imported all DLLs that it depends on, so its very similar concept here. I think everyone is using it already this way (instead of say rewriting dependent DLLs separately) so this should just work!

Reply:
Yep, we kind of need to know the entire scope of rewriting so we can know if an external type is going to be rewritten or needs to be wrapped by the InterAssemblyInvocationRewriter.